src/usage_metrics/core/zenodo.py

```diff
@@ -0,0 +1,172 @@
+            "metadata.resource_type.title",
+            "metadata.resource_type.type",
+            "metadata.license.id",
+            "links.self_html",
+            "links.doi",
+            "links.self_doi",
+            "links.self_doi_html",
+            "links.parent",
+            "links.parent_html",
+            "links.parent_doi",
+            "links.parent_doi_html",
+            "links.self_iiif_manifest",
+            "links.self_iiif_sequence",
+            "links.files",
+            "links.media_files",
+            "links.archive",
+            "links.archive_media",
+            "links.latest",
+            "links.latest_html",
+            "links.versions",
+            "links.draft",
+            "links.reserve_doi",
+            "links.access_links",
+            "links.access_grants",
+            "links.access_users",
+            "links.access_request",
+            "links.access",
+            "links.communities",
+            "links.communities-suggestions",
+            "links.requests",
+        ]
+    )
+
+    df = df.rename(
+        columns={
+            "created": "version_creation_date",  # Datetime
+            "modified": "version_last_modified_date",  # Datetime
+            "updated": "version_last_updated_date",  # Datetime
+            "metadata.publication_date": "version_publication_date",  # Date
+            "id": "version_id",
+            "recid": "version_record_id",
+            "conceptrecid": "concept_record_id",
+            "doi": "version_doi",
+            "conceptdoi": "concept_record_doi",
+            "doi_url": "version_doi_url",
+            "title": "version_title",
+            "status": "version_status",
+            "state": "version_state",
+            "submitted": "version_submitted",
+            "metadata.description": "version_description",
+            "metadata.version": "version",
+            "stats.downloads": "dataset_downloads",
+            "stats.unique_downloads": "dataset_unique_downloads",
+            "stats.views": "dataset_views",
+            "stats.unique_views": "dataset_unique_views",
+            "stats.version_downloads": "version_downloads",
+            "stats.version_unique_downloads": "version_unique_downloads",
+            "stats.version_views": "version_views",
+            "stats.version_unique_views": "version_unique_views",
+        }
+    )
+
+    # Convert string to date using Pandas
+    for col in ["metrics_date", "version_publication_date"]:
+        df[col] = pd.to_datetime(df[col])
+        df[col] = df[col].dt.date
+
+    # Convert string to datetime using Pandas
+    for col in [
+        "version_creation_date",
+        "version_last_modified_date",
+        "version_last_updated_date",
+    ]:
+        df[col] = pd.to_datetime(df[col])
+
+    # Check validity of PK column
+    df = df.set_index(["metrics_date", "version_id"])
+    assert df.index.is_unique
+
+    # Add a column with the dataset slug
+    dataset_slugs = (
```

Comment:
Non-blocking note to selves: this mapping *seems* like it should live somewhere a little bit less obscure... like in some sort of core config file or something 🤷 a problem for later.

Like maybe we should define `GenericDatasetSettings.full_human_name`... but then we'd have to import `pudl` whenever we wanted to access this, so maybe we want to finally break the datastore out... 🤷

Reply:
This is basically remapping the sources.title and sources dictionary key from `pudl.metadata.sources.py` (except with manual addition of backwards compatibility as we've changed these rarely over time), but I really wanted to avoid making this module depend on PUDL directly. It is Zenodo-specific, which is why I wanted to keep it in here rather than `helpers.py` or something. So, I'd like to punt on this one for now.